script.rb

```diff
@@ -0,0 +1,172 @@
+      end
+    else
+      var.my_select { |x| !x.nil? && x != false }.length == var.length
+    end
+  end
+
+  def my_any?(pattern = nil)
+    var = to_a
+    if block_given? == true
+      var.my_select { |x| yield(x) }.length.positive? ? true : false
+    elsif !pattern.nil?
+      if pattern.class == Regexp
+        var.my_select { |x| !pattern.match(x).nil? }.length.positive? ? true : false
+      elsif pattern.class == Integer
+        var.my_select { |x| x == pattern }.length.positive? ? true : false
+      else
+        var.my_select { |x| x.class == pattern}.length.positive? ? true : false
+      end
+    else
+      var.my_select { |x| !x.nil? && x != false }.length.positive? ? true : false
+    end
+  end
+
+  def my_none?(pattern = nil)
+    var = to_a
+    if block_given? == true
+      var.my_select { |x| yield(x) }.length.zero? ? true : false
+    elsif !pattern.nil?
+      if pattern.class == Regexp
+        var.my_select { |x| !pattern.match(x).nil? }.length.zero? ? true : false
+      elsif pattern.class == Integer
+        var.my_select { |x| x == pattern}.length.zero? ? true : false
+      else
+        var.my_select { |x| x.class == pattern}.length.zero? ? true : false
+      end
+    else
+      var.my_select { |x| !x.nil? && x != false }.length.zero? ? true : false
+    end
+  end
+
+  def my_count(arg = nil)
+    var = to_a
+    if arg.nil?
+      return var.length unless block_given?
+
+      var.my_select { |x| yield(x) }.length
+    else
+      var.my_select { |x| x == arg }.length
+    end
+  end
+
+  def my_map(proc = nil)
+    var = to_a
+    new_arr = []
+    if !proc.nil?
+      var.my_each do |x|
+        new_arr.push(proc.call(x))
+      end
+    else
+      return to_enum(:my_map) unless block_given?
+
+      var.my_each do |x|
+        new_arr.push(yield(x))
+      end
+    end
+    new_arr
+  end
+
+  def my_inject(acc = nil, sym = nil)
+    var = to_a
+
+    count = if var.my_all? { |x| x.class == String }
+              ''
+            else
+              acc
+            end
+
+    if !acc.nil? && !sym.nil?
+      operation = sym
+      var.my_each do |x|
+        count = count.send operation, x
+      end
+      count
+    elsif !acc.nil? && acc.class != Integer
+      count = 0
+      operation = acc
+      var.my_each do |x|
+        count = count.send operation, x
+      end
+      count
+    elsif acc.class == Integer
+      return 'LocalJumpError (no block given)' unless block_given?
+
+      var.my_each do |x|
+        count = yield(count, x)
+      end
+      count
+    else
+      if count.class != String
+        return 'LocalJumpError (no block given)' unless block_given?
+
+        count = 0
+        var.my_each do |x|
+          count = yield(count, x)
+        end
+        count
+      else
+        return 'LocalJumpError (no block given)' unless block_given?
+
+        var.my_each do |x|
+          count = yield(count, x)
+        end
+        count
+      end
+    end
+  end
+end
```

Comment:
Finally:

![image](https://user-images.githubusercontent.com/2739245/94214057-19d68b00-fe9e-11ea-8f4f-214d564622a7.png)

Make sure `#my_inject` raises a LocalJumpError when no block or argument is given, make sure  when a block is given without an initail value combines all elements of enum by applying a binary operation specified and finally make sure that when a symbol is specified without an initail value combines each element of the collection by applying the symbol.

To test these cases, you may use the following code:

```
      # 'combines all elements of enum by applying a binary operation, specified by a block::range'
        actual = range.my_inject { |prod, n| prod * n }
        expected = range.inject { |prod, n| prod * n }
        actual == expected

      # 'combines each element of the collection by applying the symbol as a named method:range'
        range.my_inject(:*)) == range.inject(:*)
```